language/bytecode-verifier/src/borrow_graph.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines the (acyclic) borrow graph for the type and memory safety analysis.
+use crate::nonce::Nonce;
+use mirai_annotations::checked_verify;
+use std::collections::{BTreeMap, BTreeSet};
+
+/// There are two types of edges in the borrow graph.
+/// A strong edge carries more information than a weak edge.
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+enum EdgeType {
+    Strong,
+    Weak,
+}
+
+/// The label on an edge
+pub type Label<T> = Vec<T>;
+
+/// A labeled edge in borrow graph
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
```

Comment:
This suggests to me that a Nonce can be cloned, which does not seem like a great thing to me. Alternatively, it seems like Nonce is really a misnomer.

Reply:
As I have indicated in a different comment, I am happy to change Nonce to something else.  @tnowacki used RefID to represent the same concept in the source level borrow checker. Shall I use that?

Reply:
+1 for RefID, it's not a nonce as its used more than once :)

Reply:
Nonce seems to be a node in the graph. Since the graph is concerned with borrowing and you can only borrow things that can be referenced it does make some sense to prefer a name that alludes to references. RefID is a bit clunky, but there is value in being consistent across the code base.

Reply:
ok, RefID it is.  I will make this change consistently in a subsequent PR.